internal/ics/app.go

```diff
@@ -1,58 +1,52 @@
 package ics
 
 import (
-	"os"
-
 	"github.com/spf13/cobra"
 	"github.com/wavesoftware/go-commandline"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
-	"knative.dev/kn-plugin-event/pkg/configuration"
-	"knative.dev/kn-plugin-event/pkg/system"
+	outlogging "knative.dev/client/pkg/output/logging"
+	"knative.dev/kn-plugin-event/pkg/binding"
+	"knative.dev/kn-plugin-event/pkg/cli"
+	"knative.dev/kn-plugin-event/pkg/k8s"
 	"knative.dev/pkg/logging"
 )
 
 // Options to override the commandline for testing purposes.
 var Options []commandline.Option //nolint:gochecknoglobals
 
-type App struct{}
+type App struct {
+	k8s.Params
+}
 
 func (a App) Command() *cobra.Command {
-	return &cobra.Command{
+	c := &cobra.Command{
 		Use:           "ics",
 		SilenceUsage:  true,
 		SilenceErrors: true,
 		RunE:          a.run,
 	}
+	c.SetContext(cli.InitialContext())
+	c.PersistentPreRun = func(cmd *cobra.Command, _ []string) {
+		cli.SetupContext(cmd, zapcore.DebugLevel)
```

Comment:
On K8s, the sender fails with:

```
mkdir /.cache: permission denied
```

Reply:
That's addressed in the latest change, right? https://github.com/knative-extensions/kn-plugin-event/pull/368/commits/76a4108e8bb401f336f678b34972b2b35a484583

Reply:
Yes, however, I still wonder why e2e tests haven't caught it...